gamemodes/terrortown/gamemode/server/sv_weaponry.lua

```diff
@@ -639,6 +639,11 @@ concommand.Add("ttt_dropammo", DropActiveAmmo)
 -- @ref https://wiki.garrysmod.com/page/GM/WeaponEquip
 -- @local
 function GM:WeaponEquip(wep, ply)
+	if IsValid(ply) then
+		-- allow the player to pick up another weapon
+		ply.wpickup_waitequip = false
```

Comment:
Theoretically this could be simplified because there are multiple checks of `IsValid(ply)` (nothing of that except the weapon removing should happen if ply is invalid)

Reply:
Just a small request. You could let it stay as it is, but just in case to keep the code simple :)

Reply:
It's checked twice because it happens at two different points in time. The player could be valid and suddenly become invalid during the next frame, which is why I would keep it

Reply:
Sure, but the timer shouldn't start if the player is invalid at the first time too. Do you understand what I mean? Theoretically, the weapon should get removed if it is incompatible (return ?!). After this, if the player is valid, the other stuff should run (equipping, ...). Currently it seems to be inconsistent :o

Reply:
Ah, I see now, I must have been looking at outdated files. I get your point now, Ill fix it